server/src/test/scala/controllers/application/SummariesControllerSpec.scala

```diff
@@ -0,0 +1,232 @@
+
+import akka.actor.ActorSystem
+import akka.pattern.ask
+import akka.stream.Materializer
+import drt.shared.CrunchApi.{CrunchMinute, MinutesContainer}
+import module.DRTModule
+import org.scalatest.BeforeAndAfterEach
+import org.scalatestplus.play.PlaySpec
+import play.api.mvc.{AnyContentAsEmpty, Headers}
+import play.api.test.Helpers._
+import play.api.test._
+import services.crunch.H2Tables
+import services.graphstages.Crunch
+import slick.jdbc.H2Profile.api._
+import uk.gov.homeoffice.drt.crunchsystem.DrtSystemInterface
+import uk.gov.homeoffice.drt.db.queries.PassengersHourlyDao
+import uk.gov.homeoffice.drt.ports.AirportConfig
+import uk.gov.homeoffice.drt.ports.Queues.{EeaDesk, NonEeaDesk, Queue}
+import uk.gov.homeoffice.drt.ports.Terminals.{T2, T3, Terminal}
+import uk.gov.homeoffice.drt.ports.config.Lhr
+import uk.gov.homeoffice.drt.time.{LocalDate, SDate, SDateLike}
+
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.DurationInt
+
+class SummariesControllerSpec extends PlaySpec with BeforeAndAfterEach {
+  implicit val system: ActorSystem = akka.actor.ActorSystem("test")
+  implicit val mat: Materializer = Materializer(system)
+
+  val schema = PassengersHourlyDao.table.schema
+
+  override def beforeEach(): Unit = {
+    Await.ready(H2Tables.db.run(DBIO.seq(schema.dropIfExists, schema.create)), 1.second)
+  }
+
+  def generateMinutes(start: SDateLike, end: SDateLike, terminals: Seq[Terminal], queues: Seq[Queue], paxPerHour: Double): Seq[CrunchMinute] = {
+    (start.millisSinceEpoch to end.millisSinceEpoch by 60000L)
+      .flatMap(millis => terminals.flatMap(terminal => queues.map(queue => (terminal, queue, millis))))
+      .map {
+        case (terminal, queue, millis) => CrunchMinute(terminal, queue, millis, paxPerHour, 0d, 0, 0, None, None, None, None, None, None, None)
+      }
+  }
+
+  private val queues: Seq[Queue] = Seq(EeaDesk, NonEeaDesk)
+  private val queuePaxPerMinute = 10
+  private val queuePaxPerHour = queuePaxPerMinute * 60
+  private val queuePaxPerDay = queuePaxPerHour * 24
+  private val terminalPaxPerDay = queuePaxPerDay * queues.size
+  private val terminals: Seq[Terminal] = Seq(T2, T3)
+
+  private def portPaxPerDay = terminalPaxPerDay * terminals.size
+
+  "populatePassengersForDate" should {
+    "create hourly entries from existing crunch minutes for the date requested" in {
+      val drtInterface = newDrtInterface
+      val minutes = MinutesContainer(generateMinutes(SDate("2024-05-31T23:00"), SDate("2024-06-01T22:59"), Seq(T3, T2), queues, queuePaxPerMinute))
+
+      Await.ready(drtInterface.minuteLookups.queueMinutesRouterActor.ask(minutes), 1.second)
+      val controller = newController(drtInterface)
+
+      val result = controller.populatePassengersForDate("2024-06-01").apply(FakeRequest())
+
+      status(result) must ===(OK)
+
+      val function = PassengersHourlyDao.hourlyForPortAndDate("LHR", Option("T3"))
```

Comment:
instead of function should we call it more relevant or directly use it if not called multiple time ?

Reply:
Good catch :)